### DISCUSSION TITLE
Impact of Transparent Ceilings and Objects (Null BSDF) on Coverage Maps in Indoor Scenarios

### DISCUSSION THREAD

**Discussion by steffenitely** [NVlabs/sionna#606] [Q&A] [answered]
Hi everyone,

For the visualization of coverage maps in indoor scenarios, I want to use the rendered 3D images with the coverage map (CM) and avoid using the 2D image that cm.show() gives me. Since my scene is indoors, I use transparent (null BSDF) materials for the ceiling and other objects near the ceiling.

I noticed that when opened with the preview function, I don't encounter any problems, and the CM is perfectly visible. However, when the same shot is rendered, the invisible objects near the ceiling (including the ceiling itself) obstruct the view of the CM. This makes the rendered 3D images of CMs for indoor scenarios unusable. The pictures below make it easier to see what I mean:
Preview of a custom scene with CM and invisible objects near the ceiling:
![image](https://github.com/user-attachments/assets/f7091c7c-724a-4545-b965-97df5c2e64c3)

Rendered image of the same scene:
![image](https://github.com/user-attachments/assets/40708723-8d54-494d-8370-acfbd70d71bf)

The invisible ceiling was not added in this scene, because otherwise, the whole CM would not be visible in the rendered image.

Does anyone know if there are any current workarounds or solutions for this problem?

Edit: One workaround is to first simulate the CM **with** the invisible ceiling/objects. Then load the same scene **without** ceiling/objects and finally render the CM inside the new scene in order to avoid the visual bug.

**Comment by merlinND** [comment]
Hello @steffenitely,

In the scene's XML file, do the ceiling objects that you want to make transparent actually use the `null` BSDF? If so, they should in principle not show up in the rendered image.
If not, could you please send over the full scene (scene file + mesh files) so that I can try and reproduce the issue locally.

**Reply by steffenitely** [reply]
The objects are invisible (they use a null BSDF). I believe it is a visible bug with invisible objects and the coverage map. Here are two more pictures of the same exact scene but from different angles: 
![image](https://github.com/user-attachments/assets/528d6aa0-fdc9-41cb-8bdb-fb9082f3bdb7)
![image](https://github.com/user-attachments/assets/9c791360-79e1-4d53-8c25-f2471d316c8d)

As it can be seen in the first picture the objects obstruct the view of the CM while in the second picture it can be seen that the objects are invisible and don't obstruct the view of the coverage map and therby the CM can be seen fully (only from an awkard angle).

To avoid confusion: the cell with the rendered scene in the second image was executed as the 31st cell and not 26th since i tried a couple of times to get a good angle for the render.

Since the scene is quite big i will first simplify the scene by deleting everything unneccessary and afterwards i will send the files to you.

**Reply by steffenitely** [reply]
Here are the files to reproduce the bug:
[InvisObjects_CM.zip](https://github.com/user-attachments/files/17277698/InvisObjects_CM.zip)

**Reply by merlinND** [reply, marked as answer]
Thanks for sharing the scene, now I understand the issue. I thought there was a ceiling + maybe some ceiling lamps, so I didn't understand why the ceiling worked fine but not the lamps.
The solution that's currently implemented to detect where to show the coverage map vs where to show objects that are above it does not take into account the `null` BSDF (or any BSDF).

We will try to fix this shortcoming in a future version.

**Reply by steffenitely** [reply]
That would be very helpful 
Thank you!
